packages/preview-middleware/src/base/flp.ts

```diff
@@ -227,9 +227,13 @@ export class FlpSandbox {
                 previewUrl = `${previewUrl}.inner.html`;
                 editor.pluginScript ??= 'open/ux/preview/client/cpe/init';
                 this.router.get(editor.path, (_req: Request, res: Response) => {
+                    let templatePreviewUrl = `${previewUrl}?sap-ui-xx-viewCache=false&fiori-tools-rta-mode=forAdaptation&sap-ui-rta-skip-flex-validation=true&sap-ui-xx-condense-changes=true#${this.config.intent.object}-${this.config.intent.action}`;
+                    if (rta.options?.scenario === 'ADAPTATION_PROJECT') {
+                        templatePreviewUrl = templatePreviewUrl.replace('?', `?sap-ui-layer=${rta.layer}&`);
```

Comment:
Suggestion, can't we just append 'sap-ui-layer' query param at the end of the url, since the order of the query params does not matter

Reply:
The parameters after the anchor(#) are not handled and depend on that I had two options to put it at the beginning or at the end before the anchor(#), but I chose the beginning because I think the anchor(#) can be more than one because of the app navigation

Reply:
Got it.